helpers/sortAndFilterHotels.js

```diff
@@ -0,0 +1,51 @@
+/* jslint node: true */
+'use strict';
+
+var _ = require('lodash');
+var sortBy = require('../lib/sortBy');
+
+module.exports = function(dust) {
+
+  /*
+   * @disclaimer - this is a temporary filter helper to separate a recommended hotel(s) from a list of hotels. This is temporary because it's written in dust, which will change soon.
+   *
+   * @description Augmented sorting method that orders a list of things based on a child node, and then filters the result based on an object and a value .
+   * @param {object} parent Object to sort
+   * @param {string} node Child node to sort object by
+   * @param {object} filterArray - array to base the filter on
+   * @param {boolean as string} exclude - if 'true', the filter function will exclude all items in the filterArray, based on the filterValue
+   * @param {boolean as string} include - if 'true', the filter function will include all items in the filterArray
+   * @example {@_sortAndFilterHotels parent=packageRatesReply.packageRates node="grossPrice" filterArray=_brandConfig.featuredHotel filterValue="x.links.hotelProducts.ids[0]" include="true"} {/_sortAndFilterHotels} output loop of parent object sorted by node and filtered by filterArray
+   */
+
+  dust.helpers._sortAndFilterHotels = function(chunk, context, bodies, params) {
+    var filterArray = params.filterArray;
+
+    var mappedObject = sortBy(params.parent, params.node);
+    var newMappedObject = [];
+
+    //if filterArray exists, filter out from mappedObject
+    if (filterArray.length) {
+      if (params.exclude) {
```

Comment:
There's a few lines of very similar code here, could we refactor to something like this?
```
if (params.include || params.exclude) {
	newMappedObject = _.filter(mappedObject, function(object) {
		//filter mapped object, exclude all recommended hotels from hotels list
		var hotelIndex = filterArray.indexOf(object.links.hotelProducts.ids[0]); 
		if (params.include) {
			return hotelIndex > -1;
		}
		//exclude logic...
		return hotelIndex < 0;
	});
}
```
Breaks the logic checking piece (`>` or <`) apart too